strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/storage/validation/version/NugetReleaseVersionValidator.java

```diff
@@ -10,7 +10,8 @@
 /**
  * @author Przemyslaw Fusik
  */
-@Component("NugetReleaseVersionValidator")
+//TODO: invalid implementation
+//@Component("NugetReleaseVersionValidator")
 public class NugetReleaseVersionValidator
```

Comment:
@carlspring do we need `NugetReleaseVersionValidator` in case of `SemanticVersioningValidator` ?

Reply:
NuGet doesn't understand the difference between snapshots and releases. It treats all versions as releases.

According to Semver, if you have a version that is comprised of 4 components (`1.2.3.456`), `456` is the build number. In my opinion, we should treat `456` as an analogue of Maven snapshot build numbers. So, if we see that a build number consists of four components, then it's a snapshot. What I'm not too sure about it, then... if `1.2.3.456` is considered the stable version picked for the release, is the release version then supposed to be `1.2.3`?

I think we need to come up with a solution to this sort of issue, because this could then be applicable to NPM, NuGet and all the likes which use this sort of versioning model.

What are your thoughts on this?

Reply:
Also, I agree that there should probably be just one generic implementation for this that could be configured for the different layout providers.